src/processing.integration-test.ts

```diff
@@ -81,28 +97,186 @@ describe("processSpace", () => {
         toBlock,
       }),
       emitter,
-      notifyFn,
       updateSpaceFn: updateSpace,
+      processProposalsFn,
+      processAnswersFn,
     });
 
     expect(result.lastProcessedBlock).to.equal(toBlock);
-    expect(notifyFn.calledOnce).to.be.true;
-    const call = notifyFn.firstCall.firstArg;
-    expect(call).to.deep.equal({
-      type: EventType.NEW_ANSWER,
-      space,
-      event: {
+
+    expect(processProposalsFn.calledOnce).to.be.true;
+    const processProposalsCall = processProposalsFn.firstCall;
+    expect(processProposalsCall.args[1]).to.have.lengthOf(0);
+
+    expect(processAnswersFn.calledOnce).to.be.true;
+    const processAnswersCall = processAnswersFn.firstCall;
+    expect(processAnswersCall.args[0]).to.deep.equal(space);
+    expect(processAnswersCall.args[1]).to.deep.equal([
+      {
         questionId: "0x8566ba6b1ac945f2b152a20ecc7cb3a87982190190af14cb4fbc85e12eb474e2",
         answer: "0x0000000000000000000000000000000000000000000000000000000000000001",
         user: "0x4D6CAa3E0983fAc7B514D60339EBb538C5A85AAe",
         bond: 10000000n,
-        ts: 1712934227,
         txHash: "0x0cc20c32ee428bdb8f16fa1aa22b396ecafa91b61bc2c3350723e4dfefeebff0",
         blockNumber: 19640300n,
+        happenedAt: new Date(1712934227 * 1000),
       },
-    } as Notification);
+    ]);
 
     const [updatedSpace] = await findSpaces([space.ens]);
     expect(updatedSpace.lastProcessedBlock).to.equal(toBlock);
   });
-});
\ No newline at end of file
+});
+
+describe("processProposals", () => {
+  const fn = configurableProcessProposals;
+
+  const space = randomizeSpace();
+
+  beforeEach(async () => {
+    space.ens = randomizeEns();
+    await insertSpaces([
+      {
+        ens: space.ens,
+        startBlock: 1n,
+        lastProcessedBlock: 1n,
+      },
+    ]);
+  });
+
+  it("should register the proposal and issue a notification", async () => {
+    const notifyFn = spy();
+
+    const proposal: ProposalQuestionCreated = {
+      proposalId: "0x791b3d71ea14497d3b8e756479f6f126e08b44351bfab904834c56b1ccf0479a",
+      questionId: "0xebf5b601fedfaa5562a03590e9ac8be937cc070a131443af01948a7eda6dfabf" as Hash,
+      txHash: "0x890ddd7826fcd79ff17b54368e8df393959f269847ceeb0fea13cc4b68330d43",
+      blockNumber: 19475120n,
+      happenedAt: new Date("2024-03-20T09:48:23.000Z"),
+    };
+
+    const spaceWithRealOracle = {
+      ...space,
+      oracleAddress: ONEINCH_ORACLE_ADDRESS,
```

Comment:
We need the tests to use the real oracle address to correctly resolve the related `LogNewQuestion` event.